Lib/http/server.py

```diff
@@ -686,10 +687,30 @@ def send_head(self):
         except OSError:
             self.send_error(HTTPStatus.NOT_FOUND, "File not found")
             return None
+
+        fs = os.fstat(f.fileno())
+            
         try:
+            # Use browser cache if possible
+            if ("If-Modified-Since" in self.headers
+                    and "If-None-Match" not in self.headers):
+                # compare If-Modified-Since and time of last file modification
+                ims = email.utils.parsedate_to_datetime(
+                    self.headers["If-Modified-Since"])
+                if (ims is not None and
+                        ims.tzinfo is datetime.timezone.utc):
```

Comment:
This should not happen often, but it is supported in commit c3337f40207a402944b61e89b617b21599b0519e